pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -86,10 +86,12 @@ func (w *PodWatcher) computeChanges(podList []*Pod) ([]*Pod, error) {
 			// We don't check container readiness as init containers are never ready
 			// We check if the container has an ID instead (has run or is running)
 			if !container.IsPending() {
-				if _, found := w.lastSeen[container.ID]; found == false {
+				// We store readiness in the cache key to resubmit the container on pod phase change
+				containerCacheKey := container.ID + "-" + strconv.FormatBool(IsPodReady(pod))
```

Comment:
maybe adding "-ready" at the end instead of "-true" or "-false" is more readable for a human in case of problem. But we don't often look at the cache-key :)